### ISSUE TITLE
parsing segment "staticMicrosoftOperationalInsights": expected the segment "microsoft.operationalinsights" to be "Microsoft.OperationalInsights"

### ISSUE THREAD

**Issue by AndreTheHunter** [hashicorp/terraform-provider-azurerm#18073]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.7

### AzureRM Provider Version

3.19.1

### Affected Resource(s)/Data Source(s)

azurerm_log_analytics_workspace

### Terraform Configuration Files

```hcl
Confidential
```

### Debug Output/Panic Output

```shell
2022-08-23T09:10:10.535+1000 [ERROR] provider.terraform-provider-azurerm_v3.19.1_x5: Response contains error diagnostic: @caller=github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/diag/diagnostics.go:56 @module=sdk.proto diagnostic_severity=ERROR diagnostic_summary="parsing "/subscriptions/be2c045d-bc3b-4a4d-839c-7dfcb1bdbfbf/resourceGroups/aeLogging/providers/microsoft.operationalinsights/workspaces/jesiae-logging-stag": parsing segment "staticMicrosoftOperationalInsights": expected the segment "microsoft.operationalinsights" to be "Microsoft.OperationalInsights"" tf_proto_version=5.2 tf_resource_type=azurerm_log_analytics_workspace diagnostic_detail= tf_provider_addr=provider tf_req_id=a71f9d19-a8a7-f1be-4994-9e5c139365bc tf_rpc=ReadResource timestamp=2022-08-23T09:10:10.535+1000
2022-08-23T09:10:10.535+1000 [ERROR] vertex "azurerm_log_analytics_workspace.log_analytics_workspace" error: parsing "/subscriptions/be2c045d-bc3b-4a4d-839c-7dfcb1bdbfbf/resourceGroups/aeLogging/providers/microsoft.operationalinsights/workspaces/jesiae-logging-stag": parsing segment "staticMicrosoftOperationalInsights": expected the segment "microsoft.operationalinsights" to be "Microsoft.OperationalInsights"
2022-08-23T09:10:10.535+1000 [ERROR] vertex "azurerm_log_analytics_workspace.log_analytics_workspace" error: parsing "/subscriptions/be2c045d-bc3b-4a4d-839c-7dfcb1bdbfbf/resourceGroups/aeLogging/providers/microsoft.operationalinsights/workspaces/jesiae-logging-stag": parsing segment "staticMicrosoftOperationalInsights": expected the segment "microsoft.operationalinsights" to be "Microsoft.OperationalInsights"
2022-08-23T09:10:10.535+1000 [ERROR] vertex "azurerm_log_analytics_workspace.log_analytics_workspace (expand)" error: parsing "/subscriptions/be2c045d-bc3b-4a4d-839c-7dfcb1bdbfbf/resourceGroups/aeLogging/providers/microsoft.operationalinsights/workspaces/jesiae-logging-stag": parsing segment "staticMicrosoftOperationalInsights": expected the segment "microsoft.operationalinsights" to be "Microsoft.OperationalInsights"
```

### Expected Behaviour

Plan should work without error

### Actual Behaviour

```
│ Error: parsing "/subscriptions/41e444de-4c1e-4bf8-aaad-c5f5d5a3b71b/resourceGroups/aeLogging/providers/microsoft.operationalinsights/workspaces/jesiae-logging-stag": parsing segment "staticMicrosoftOperationalInsights": expected the segment "microsoft.operationalinsights" to be "Microsoft.OperationalInsights"
│
│   with azurerm_log_analytics_workspace.log_analytics_workspace,
│   on logging.tf line 26, in resource "azurerm_log_analytics_workspace" "log_analytics_workspace":
│   26: resource "azurerm_log_analytics_workspace" "log_analytics_workspace" {
```

Downgrading `hashicorp/azurerm` to `3.18.0` fixes the issue.

### Steps to Reproduce

```sh
terraform plan -refresh=true
```

**Comment by magodo** [comment]
@AndreTheHunter This error implies that the id of `azurerm_log_analytics_workspace.log_analytics_workspace` is of incorrect casing for the `Microsoft.OperationalInsights` segment. Would you please confirm this via the output of `terraform show`?

This usually won't happen as there is a state migration that will handle this. Would you also check the `terraform.tfstate` file to see what the schema version is:

```
cat terraform.tfstate | jq '.resources | .[] | select((.type == "azurerm_log_analytics_workspace") and .name == "log_analytics_workspace") | .instances | .[] | .schema_version'
```

The expected value should be 2. If this is the case, then could you check whether this resource is imported prior to version v3.19.0? In this case, you shall remove it from state and import again, and make sure during importing, the casing of these fixed segments match with what is defined at: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/log_analytics_workspace#import

**Comment by myc2h6o** [comment]
There is an sdk upgrade in #18098 which enhanced the ID validation, thus a resource imported previously with wrong ID will fail when running `terraform plan`. This shall have been fixed in #20240 which migrates the ID automatically.

**Comment by tombuildsstuff** [comment]
Closing since per the above comment this has been fixed, if this is still an issue in the latest version of the Provider then please let us know and we can take another look.

Thanks!
